unit_tests/ex_suffix_expander.cc

```diff
@@ -9,13 +9,177 @@
 */
 
 #include <unordered_map>
+#include <cctype>    // isdigit, isspace, tolower
+#include <algorithm> // std::equal
+#include <iostream>  // debug (TODO: remove)
 
 #include "swoc/TextView.h"
 #include "swoc/swoc_file.h"
 #include "catch.hpp"
 
 using swoc::TextView;
 
+// TODO: some comments here stating the assumptions I've made
+
+// The Metric concept
+// A metric is a measurement system which can be computer storage, time duration
+// etc.
+/*
+struct Metric {
+  // the base (smallest) unit of the metric system
+  static constexpr TextView baseUnit = "";
+
+  // convert different units to a multiply of the base unit
+  static inline uintmax_t convertToBase(TextView const unit);
+
+  // canonicalize the spelling of the metric. e.g. both "b" and "byte" mean the
+  // same thing
+  // intentionally *not* taking a const unit b/c there might be use cases where
+  // the implementation wants to modify the view and then return it
+  static inline TextView canonicalize(TextView unit);
+};
+*/
+
+inline bool iequal(TextView const lhs, TextView const rhs) noexcept {
```

Comment:
Not needed - there is an overload of `strcasecmp` that does this.

Reply:
I'm always in awe of the c string API thinking "the null terminator is going to assassinate me!" :-{
Just looked at the API, looks like you're alluding to the `int strncasecmp(const char *s1, const char *s2, size_t n);` overload and I should pass in TextView/string_view::data()? but what if lhs is longer than rhs, but the implementation doesn't know it needs to stop after it reaches position=rhs.size() since there is no null terminator?

I guess it should be:
```cpp
inline bool iequal(TextView const lhs, TextView const rhs) noexcept {
  return lhs.size() == rhs.size() && strncasecmp(lhs.data(), rhs.data(), lhs.size();
}
```
Please let me know if my understanding are correct :-]

Reply:
No, there's an overload of the function that understands `TextView`. E.g. you can do
```
bool same(TextView lhs, TextView rhs) { return 0 == strcasecmp(lhs, rhs); }
```
See here - https://github.com/SolidWallOfCode/libswoc/blob/training/code/include/swoc/TextView.h#L24

Reply:
ack